### PR TITLE
Add missing TEST_OUTPUT sections to most tests in compilable

### PR DESCRIPTION
**Pull request by MoonlightSentinel**
These test were simply missing an appropriate section. The remaining ones need
some extensions to d_do_test.d to handle e.g. paths.

~~Currently includes #10754 to avoid some deprecations messages regarding `body`.~~

### CODE REVIEW

File: test/compilable/staticforeach.d
Comment (Geod24) [inline test/compilable/staticforeach.d]: This gives `0u` & co on 32 bits machines, so I'm assuming there's a sneaky `size_t` in there you need to replace.

Comment (MoonlightSentinel) [reply test/compilable/staticforeach.d]: Missed some loop index variables, should be fixed now

File: test/compilable/staticforeach.d
Comment (Geod24) [inline test/compilable/staticforeach.d]:
```suggestion
            mixin("enum int[] "~name~name2~"=[cast(int) i, cast(int) j];");
```

File: test/compilable/staticforeach.d
Comment (Geod24) [inline test/compilable/staticforeach.d]:
```suggestion
    pragma(msg, int(i)," ",j);
```

File: test/compilable/staticforeach.d
Comment (Geod24) [inline test/compilable/staticforeach.d]:
```suggestion
        pragma(msg, int(i)," ",v);
```

File: test/compilable/test7172.d
Comment (Geod24) [inline test/compilable/test7172.d]: You can just remove the class, that's not what the issue was about: https://issues.dlang.org/show_bug.cgi?id=7172

File: test/compilable/test8509.d
Comment (Geod24) [inline test/compilable/test8509.d]: Likewise, replace with new operator syntax, the issue wasn't about operators: https://issues.dlang.org/show_bug.cgi?id=8509

**Review by Geod24 — COMMENTED** [review]
A few changes needed, otherwise looks good, thanks!

File: test/compilable/staticforeach.d
Comment (Geod24) [inline test/compilable/staticforeach.d]: Oh I missed this one earlier

File: test/compilable/test7172.d
Comment (Geod24) [inline test/compilable/test7172.d]: Hum sorry, this deprecation has actually been reverted (in stable), so it's probably better to wait for the merge and revert this.

Comment (MoonlightSentinel) [reply test/compilable/test7172.d]: Reverted the changes to this file to unblock this PR.  This test will be fine once `TEST_OUTPUT` implies no ouput because the deprecation warnings will disappear with the stable merge.

**Review by wilzbach — APPROVED** [review]

**Review by Geod24 — APPROVED** [review]

**Pull request merged**
